### ISSUE TITLE
Unnecessary `Plugin file not found` error while registering plugin by using `just`

### ISSUE THREAD

**Issue by hustcer** [nushell/nushell#4406]
### Describe the bug

I'm using [Just](https://github.com/casey/just) to manage all my nushell scripts. While registering plugin by  invoking `just` recipe got an error output like this:
```
Error: nu::shell::plugin_failed_to_load (link)

  × Plugin failed to load: Plugin file not found
```
However the plugin actually registered successfully.

### How to reproduce

1. Install brew by `brew install just`
2. Create a `Justfile` in your `~/abc` directory with the following content:
```
set shell := ['nu', '-c']
test:
  @source ~/abc/common.nu;
```
3. Create `~/abc/common.nu` file with contents:
```
register -e capnp ~/.cargo/bin/nu_plugin_query
version
```
4. Go to `~/abc` directory and run `just test` will get the following output:
```
Error: nu::shell::plugin_failed_to_load (link)

  × Plugin failed to load: Plugin file not found

╭────────────────────┬──────────────────────────────────────────╮
│ version            │ 0.59.0                                   │
│ branch             │ main                                     │
│ short_commit       │ c2118e75                                 │
│ commit_hash        │ c2118e7505d779103bf6f6c845ea86f8a1cd70c0 │
│ commit_date        │ 2022-02-10 02:24:29 +00:00               │
│ build_os           │ macos-x86_64                             │
│ rust_version       │ rustc 1.58.1 (db9d1b20b 2022-01-20)      │
│ rust_channel       │ stable-x86_64-apple-darwin               │
│ cargo_version      │ cargo 1.58.0 (f01b232bc 2022-01-19)      │
│ pkg_version        │ 0.59.0                                   │
│ build_time         │ 2022-02-10 12:07:35 +08:00               │
│ build_rust_channel │ release                                  │
│ features           │ dataframe, default, which, zip           │
│ installed_plugins  │ query, query json, query web, query xml  │
╰────────────────────┴──────────────────────────────────────────╯
```

### Expected behavior

Why raise the following error msg? Actually, the plugin registered successfully, we can know this from `version` output, BTW, which plugin not found? The message is not detail enough

```
Error: nu::shell::plugin_failed_to_load (link)

  × Plugin failed to load: Plugin file not found
```

The expected output should be same as running `source ~/abc/common.nu`

### Configuration

| key                | value                                               |
| ------------------ | --------------------------------------------------- |
| version            | 0.59.0                                              |
| branch             | main                                                |
| short_commit       | c2118e75                                            |
| commit_hash        | c2118e7505d779103bf6f6c845ea86f8a1cd70c0            |
| commit_date        | 2022-02-10 02:24:29 +00:00                          |
| build_os           | macos-x86_64                                        |
| rust_version       | rustc 1.58.1 (db9d1b20b 2022-01-20)                 |
| rust_channel       | stable-x86_64-apple-darwin                          |
| cargo_version      | cargo 1.58.0 (f01b232bc 2022-01-19)                 |
| pkg_version        | 0.59.0                                              |
| build_time         | 2022-02-10 12:07:35 +08:00                          |
| build_rust_channel | release                                             |
| features           | dataframe, default, which, zip                      |
| installed_plugins  | gstat, inc, query, query json, query web, query xml |

**Comment by hustcer** [comment]
Sure, i will try it later

**Comment by hustcer** [comment]
```
〉nu -c 'register -e capnp ~/.cargo/bin/nu_plugin_query'                                                                                                        02/11/2022 09:40:42 AM
Error: nu::shell::plugin_failed_to_load (link)

  × Plugin failed to load: Plugin file not found
```
@fdncred It's a much simpler way to reproduce the bug

**Comment by hustcer** [comment]
Any way to suppress the output? I tried:
`nu -c 'do -i {register -e capnp ~/.cargo/bin/nu_plugin_extra_query}'`
`nu -c 'do -i {register -e capnp ~/.cargo/bin/nu_plugin_extra_query} | ignore'`
but all of them make no difference

**Comment by fdncred** [comment]
related #4316

**Comment by hustcer** [comment]
Can this issue has higher priority？It's really a nagging message. Thanks.
```
$> nu -c 'register -e capnp ~/.cargo/bin/nu_plugin_extra_query'
Error: nu::shell::plugin_failed_to_load (link)

  × Plugin failed to load: Plugin file not found

```
```
nu -c 'do -i {register -e capnp ~/.cargo/bin/nu_plugin_extra_query}'
Error: nu::shell::plugin_failed_to_load (link)

  × Plugin failed to load: Plugin file not found
```
I'm already use the latest main:
```
| key                | value                                               |
| ------------------ | --------------------------------------------------- |
| version            | 0.59.0                                              |
| branch             | main                                                |
| short_commit       | 9888f8f2                                            |
| commit_hash        | 9888f8f298e18ca60e713637914014d8767e077c            |
| commit_date        | 2022-02-21 22:22:21 +00:00                          |
| build_os           | macos-x86_64                                        |
| rust_version       | rustc 1.58.1 (db9d1b20b 2022-01-20)                 |
| rust_channel       | stable-x86_64-apple-darwin                          |
| cargo_version      | cargo 1.58.0 (f01b232bc 2022-01-19)                 |
| pkg_version        | 0.59.0                                              |
| build_time         | 2022-02-22 09:51:58 +08:00                          |
| build_rust_channel | release                                             |
| features           | dataframe, default, trash, which, zip               |
| installed_plugins  | gstat, inc, query, query json, query web, query xml |
```

**Comment by elferherrera** [comment]
Can you try with the full path for your plugin?

**Comment by hustcer** [comment]
> Can you try with the full path for your plugin?

Does not work either:

<img width="1124" alt="Screen Shot 2022-02-22 at 22 10 09" src="https://user-images.githubusercontent.com/1161607/155149330-3b2e400b-9f51-4711-aca1-5e125f02cd57.png">

**Comment by hustcer** [comment]
The plugin actually registered successfully, the warning message should not appear, just remove the error message is enough @elferherrera

**Comment by elferherrera** [comment]
When you say registered, you mean the plugin signature has been added to the plugin file. Is that correct?

**Comment by hustcer** [comment]
I mean after that message I can use the service the plugin provided, eg: `query json` and so on. If I remove the register line `query json` was not available

**Comment by hustcer** [comment]
BTW: If I run `register -e capnp ~/.cargo/bin/nu_plugin_extra_query` without `nu -c` everything is fine.

**Comment by hustcer** [comment]
Will this issue be fixed before 0.60 beta release?
